.github/workflows/storybook-deploy.yml

```diff
@@ -0,0 +1,40 @@
+name: Build and Deploy Storybook
+
+on:
+  push:
+    branches:
+      - main
+  workflow_call:
+    secrets:
+      PUBLISH_STORYBOOK_TOKEN:
+        required: true
+
+jobs:
+  build-deploy-storybook:
+    name: Build and Deploy Storybook
+    permissions:
+      contents: write
+    runs-on: ubuntu-latest
+    environment: github-pages
+    defaults:
+      run:
+        working-directory: apps/storybook-react
+    steps:
+      - name: Checkout and setup environment
+        uses: MetaMask/action-checkout-and-setup@v1
+        with:
+          is-high-risk-environment: false
+          cache-node-modules: true
+
+      - name: Install dependencies
+        run: yarn --immutable
+
+      - name: Build Storybook
+        run: yarn build-storybook
+
+      - name: Deploy to GitHub Pages
+        uses: peaceiris/actions-gh-pages@de7ea6f8efb354206b205ef54722213d99067935
```

Comment:
This action is used by a [few MM repos](https://github.com/search?q=org%3AMetaMask%20peaceiris%2Factions-gh-pages%40de7ea6f8efb354206b205ef54722213d99067935&type=code) to publish to github pages